indexed-file-analyzer/src/main/kotlin/com/avail/tools/fileanalyzer/IndexedFileAnalyzer.kt

```diff
@@ -310,11 +309,65 @@ object IndexedFileAnalyzer
 			// and ensure that any supplied paths are syntactically valid.
 			configuration = configure(args)
 			with(configuration) {
+				if (implodeDirectory !== null)
+				{
+					assert(inputFile === null)
+					assert(implodeHeader !== null)
+					assert(implodeOutput !== null)
+					val dir = implodeDirectory!!
+					// The default option file is used as the output.
+					val outputFile = implodeOutput!!
+					if (outputFile.exists())
+					{
+						throw Exception(
+							"Output file must not already exist")
+					}
+					val files = dir.list()!!.toMutableList()
+					val metadata = when
+					{
+						files.remove("metadata") -> "metadata"
+						files.remove("metadata.txt") -> "metadata.txt"
+						else -> null
+					}
+					val regex = "^(\\d+)(\\.txt)?$".toRegex(IGNORE_CASE)
+					files.forEach {
+						if (!regex.matches(it))
+						{
+							throw Exception(
+								"Unexpected file '$it'. Implode directory "
+									+ "entries must be numeric, or numeric + "
+									+ "'.txt', and contiguous, starting at 0. "
+									+ "'metadata' or 'metadata.txt' is also "
+									+ "supported.")
+						}
+					}
+					files.sortBy { regex.replace(it, "$1").toInt() }
+					for (i in files.indices)
+					{
+						val name = files[i]
+						if (regex.replace(name, "$1").toInt() != i)
+						{
+							throw Exception(
+								"Cannot find file '$i' or '$i.txt'.")
+						}
+					}
```

Comment:
Again, the code isn't defended against e.g. the existence of both `1` and `1.txt`. We should protect against this.

Reply:
Both this and the previous problem would have been reported in a different way.  Now they produce specific messages.